src/btree/bt_sync.c

```diff
@@ -385,10 +385,15 @@ __sync_page_skip(
         return (0);
 
     /*
-     * Skip reading the pages that are normal leaf pages or don't have an aggregated durable stop
-     * timestamp.
+     * Skip reading the on-disk page when it meets any of the following conditions.
+     * 1. Leaf pages with no overflow keys/values.
+     * 2. Internal pages from non-logged tables with no aggregated stop timestamp.
+     *
+     * FIXME: Read internal pages from non-logged tables when the remove/truncate
+     * operation is performed using no timestamp.
      */
-    if (addr.type == WT_ADDR_LEAF_NO || addr.ta.newest_stop_durable_ts == WT_TS_NONE) {
+    if (addr.type == WT_ADDR_LEAF_NO ||
+      (!F_ISSET(S2BT(session), WT_BTREE_LOGGED) && addr.ta.newest_stop_durable_ts == WT_TS_NONE)) {
```

Comment:
so this is more specific to oplog case.

Reply:
Yes, that's correct. Logged tables do not use timestamps, so we want to fix the tables that do not use the timestamps.